woocommerce/woocommerce-ios
# [Mobile Payments] Pass customer email when creating a payment intent

Closes #4129 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader and uploading a pre-release version of WCPay to a test site⚠️

## Changes
* Pass customer email when initiating a payment capture, as discussed in p91TBi-55F-p2#comment-4515

## How to test
**TL:DR; Add a breakpoint to StripeCardReaderService in line 236. The breakpoint should be hit when we start capturing a payment, and a `po parameters` should show the customer email address as part of the payload we pass to the Stripe Terminal SDK.**

Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. The pre-release version of WCPay that implements the endpoint that captures the payment id (see p91TBi-54R#comment-4441-p2 for a downloadable zip)
3. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* Tap the "Email Receipt" button
* The standard iOS email composing sheet is presented. If there is a customer email address on file, it should be pre-populated with it. The subject should read "Your receipt from STORE_NAME". The receipt should contain, at this point, a list of the order items, the total, and two fields we need to include into the receipt for regulatory reasons.



Update release notes:
